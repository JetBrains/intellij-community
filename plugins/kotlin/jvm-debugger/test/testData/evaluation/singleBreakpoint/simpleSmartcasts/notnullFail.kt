// Copyright 2000-2022 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
package notnullFail

fun main(args: Array<String>) {
    test(null)
}

fun test(nullable: String?) =
    //Breakpoint!
    nullable != null && nullable.length == 2

// EXPRESSION: nullable.length
// RESULT: java.lang.NullPointerException
