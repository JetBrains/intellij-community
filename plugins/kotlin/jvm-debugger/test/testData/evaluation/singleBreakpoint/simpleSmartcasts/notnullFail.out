LineBreakpoint created at notnullFail.kt:10
Run Java
Connected to the target VM
notnullFail.kt:10
Compile bytecode for nullable.length
Disconnected from the target VM

Process finished with exit code 0
